hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -333,27 +337,80 @@ public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 			final var fromThisNftsOwned = (long) tokenRelsLedger.get(fromRel, TOKEN_BALANCE);
 			final var toNftsOwned = (long) accountsLedger.get(to, NUM_NFTS_OWNED);
 			final var toThisNftsOwned = (long) tokenRelsLedger.get(asTokenRel(to, nftType), TOKEN_BALANCE);
-			nftsLedger.set(nftId, OWNER, EntityId.fromGrpcAccountId(to));
+
+			if (isKnownTreasury(to)) {
```

Comment:
This is not correct. `to` can be a treasury but it might not be treasury for this specific token that is being transferred